pkg/graveler/graveler.go

```diff
@@ -1640,31 +1640,55 @@ func (g *Graveler) Get(ctx context.Context, repository *RepositoryRecord, ref Re
 		return nil, err
 	}
 
+	var updatedValue *Value
 	if reference.StagingToken != "" {
 		// try to get from staging, if not found proceed to committed
-		value, err := g.getFromStagingArea(ctx, reference.Branch, key)
+		updatedValue, err = g.getFromStagingArea(ctx, reference.Branch, key)
 		if err != nil && !errors.Is(err, ErrNotFound) {
 			return nil, err
 		}
-		if err == nil {
-			if value == nil {
-				// tombstone - the entry was deleted on the branch => doesn't exist
-				return nil, ErrNotFound
-			}
-			return value, nil
+		// tombstone - the entry was deleted on the branch => doesn't exist
+		if err == nil && updatedValue == nil {
+			return nil, ErrNotFound
+		}
+	}
+
+	if updatedValue == nil && reference.CompactedBaseMetaRangeID != "" {
```

Comment:
How do we get here with `updatedValue != nil`?

Reply:
in case the object didn't change in staging